File: src/types/tools.ts
export interface DiskHealth {
  name: string
  model: string
  serial: string
  firmware: string
  interface: string
  capacity_bytes: number
  temperature_c: number | null
  power_on_hours: number | null
  health_status: string
  health_percent: number
  smart_attributes: SmartAttribute[]
}

export interface SmartAttribute {
  id: number
  name: string
  value: number
  worst: number
  threshold: number
  raw_value: string
  status: string
}

export interface GpuRenameInfo {
  gpu_index: number
  original_name: string
  current_name: string
  is_renamed: boolean
  registry_path: string
}

export interface NvidiaDriverInfo {
  installed_version: string
  driver_date: string
  gpu_name: string
  cuda_version: string
  latest_version: string | null
  update_available: boolean
}

export interface BuiltinTool {
  id: string
  name: string
  description: string
  category: string
  command: string
  icon: string
  requires_admin: boolean
}

File: src-tauri/src/models/tools.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DiskHealth {
    pub name: String,
    pub model: String,
    pub serial: String,
    pub firmware: String,
    pub interface: String,
    pub capacity_bytes: u64,
    pub temperature_c: Option<u32>,
    pub power_on_hours: Option<u64>,
    pub health_status: String,
    pub health_percent: u8,
    pub smart_attributes: Vec<SmartAttribute>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SmartAttribute {
    pub id: u8,
    pub name: String,
    pub value: u32,
    pub worst: u32,
    pub threshold: u32,
    pub raw_value: String,
    pub status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GpuRenameInfo {
    pub gpu_index: u32,
    pub original_name: String,
    pub current_name: String,
    pub is_renamed: bool,
    pub registry_path: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NvidiaDriverInfo {
    pub installed_version: String,
    pub driver_date: String,
    pub gpu_name: String,
    pub cuda_version: String,
    pub latest_version: Option<String>,
    pub update_available: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BuiltinTool {
    pub id: String,
    pub name: String,
    pub description: String,
    pub category: String,
    pub command: String,
    pub icon: String,
    pub requires_admin: bool,
}
